[PB-4355/PB-4354]: feat/apply taxes to the products

xabg2 opened the pull request:
## Description

Now, when the user goes to pay in the integrated checkout, he will see the breakdown of the payment to be made:
- Base price of the product
- VAT of the product depending on the user's location.

We have also updated the checkout endpoints (in the SDK) to be able to perform this task.

⚠️  There is a new env variable we must add:
`REACT_APP_LOCATION_API_URL`

## Related Issues
-

## Related Pull Requests
-

## Checklist

- [ ] Changes have been tested locally.
- [ ] Unit tests have been written or updated as necessary.
- [ ] The code adheres to the repository's coding standards.
- [ ] Relevant documentation has been added or updated.
- [ ] No new warnings or errors have been introduced.
- [ ] SonarCloud issues have been reviewed and addressed.
- [ ] QA Passed

## How Has This Been Tested?

Select a plan to purchase > see the taxes applied on the product sheet > purchase the item > check that the total payment should be the base price + taxes.

## Additional Notes

<!-- Include any additional context, potential impacts, or implementation details that reviewers should be aware of. -->

xabg2 committed:
feat(payments): apply taxes to the products

xabg2 committed:
chore(env): add new env to the example file

xabg2 committed:
Merge branch 'master' into feat/external-taxes-for-prices-in-checkout

xabg2 committed:
feat(payments): add translations to JSON

xabg2 committed:
fix(checkout): calculate the total price correctly for b2b plans

xabg2 committed:
Merge branch 'master' into feat/external-taxes-for-prices-in-checkout

CandelR commented on src/app/core/factory/sdk/index.ts:
```diff
@@ -130,6 +131,17 @@ export class SdkFactory {
+
+    const newToken = SdkFactory.sdk.localStorage.get('xNewToken');
+
+    const apiSecurity = { ...this.getApiSecurity(), token: newToken ?? '' };
```
it exists `getNewApiSecurity` that provides `newToken` directly

CandelR commented on src/app/utils/userLocationUtils.ts:
```diff
@@ -0,0 +1,13 @@
+const LOCATION_API = process.env.REACT_APP_LOCATION_API_URL;
+
+export const getUserLocation = async (): Promise<{ ip: string; location: string }> => {
+  const response = await fetch(`${LOCATION_API}`, {
```
use axios instead of fetch. Why didn't have added this call to sdk as other ones?

xabg2 replied:
Fetch adds the auth headers automatically and we don't need them to call this endpoint. 

Regarding the SDK, I can add it, yes.

CandelR commented on src/app/payment/views/IntegratedCheckoutView/CheckoutViewWrapper.tsx:
```diff
@@ -216,7 +197,7 @@ const CheckoutViewWrapper = () => {
       .then((stripe) => (stripeSdk = stripe))
       .catch((error) => {
-        errorService.reportError(error);
+        console.log(`GET STRIPE ERROR: ${error}`);
```
forgot to remove this log :)

CandelR commented on src/app/payment/views/IntegratedCheckoutView/CheckoutViewWrapper.tsx:
```diff
@@ -225,20 +206,20 @@ const CheckoutViewWrapper = () => {
       })
       .catch((error) => {
-        errorService.reportError(error);
+        console.log(`FETCH SELECTED PLAN ERROR: ${error}`);
```
same with this log

CandelR commented on src/app/payment/views/IntegratedCheckoutView/CheckoutViewWrapper.tsx:
```diff
@@ -225,20 +206,20 @@ const CheckoutViewWrapper = () => {
     handleFetchSelectedPlan(planId, currencyValue)
-      .then(async (plan) => {
-        if (checkoutTheme && plan) {
+      .then(async (price) => {
```
I think it would be better to change the name of this variable, as `price.price` is not very intuitive, you can also add the type of the argument

CandelR commented on src/app/payment/views/IntegratedCheckoutView/CheckoutViewWrapper.tsx:
```diff
@@ -259,11 +240,40 @@ const CheckoutViewWrapper = () => {
+          setPlan(price);
+        })
+        .catch(() => {
+          // Handle error
```
remove this comment, add the desired behaviour if `getPriceById` fails

CandelR commented on src/app/payment/views/IntegratedCheckoutView/CheckoutViewWrapper.tsx:
```diff
@@ -259,11 +240,40 @@ const CheckoutViewWrapper = () => {
+        }
+      })
+      .catch(() => {
+        console.log('ERROR GETTING USER LOCATION');
```
Same here, add the desired behaviour if `getUserLocation` fails

CandelR commented on src/app/payment/views/IntegratedCheckoutView/CheckoutViewWrapper.tsx:
```diff
@@ -493,6 +504,18 @@ const CheckoutViewWrapper = () => {
+  ) => {
+    const price = await checkoutService.getPriceById({ priceId, currency, promoCodeName, postalCode, country });
+    setSelectedPlan(price);
+    setPlan(price);
```
It is a bit confusing see `price` named variable adding it to a `plan` ones 😵‍💫

xabg2 replied:
Yes, I know. We have to remove useless code in general... It was used when we had upsell plans, but since we only have yearly plans, we don't need them anymore... anyways, I will try to remove this state.

CandelR reviewed (commented):
Remember to add the appropriate tests and check sonar cloud new issues

xabg2 committed:
fix(checkout): remove useless logs and useless states

This commit removes useless logs and states. This commit also includes an `envConfig` file so we can centralize all env variables in a single file instead of having them throughout the code.

xabg2 committed:
chore(deps): update SDK version

xabg2 committed:
tests(checkout): add coverage for checkout endpoints

xabg2 committed:
test(user): remove useless test

xabg2 committed:
tests(checkout): create payment intent

xabg2 committed:
tests(plans): fetch plans

xabg2 committed:
tests: fix description

xabg2 committed:
tests(checkout): add coverage to checkout service

xabg2 committed:
feat(checkout): get price by user address data

xabg2 committed:
chore(deps): bump SDK version

xabg2 committed:
fix: undo some inneccessary changes

xabg2 committed:
fix: undo some changes in translations

xabg2 committed:
feat(checkout): format prices correctly

xabg2 committed:
tests(checkout): add coverage to checkout service

xabg2 committed:
tests(checkout): add coverage for checkout service

xabg2 committed:
tests(checkout): add coverage when loading Stripe Elements

xabg2 committed:
tests(checkout): add coverage to useCheckout custom hook

xabg2 committed:
fix(plans): use nullish coalesing instead

xabg2 committed:
fix(actions): add missing env variable to playwright Action
